EIPS/eip-7543.md

```diff
@@ -0,0 +1,158 @@
+
+
+<!--
+  This section is optional.
+
+  The motivation section should include a description of any nontrivial problems the EIP solves. It should not describe how the EIP solves those problems, unless it is not immediately obvious. It should not describe why the EIP should be made into a standard, unless it is not immediately obvious.
+
+  With a few exceptions, external links are not allowed. If you feel that a particular resource would demonstrate a compelling case for your EIP, then save it as a printer-friendly PDF, put it in the assets folder, and link to that copy.
+
+  TODO: Remove this comment before submitting
+-->
+
+## Specification
+
+### Decimal
+
+A decimal is defined as
+
+c * 10^q
+
+where c and q are int256.
+
+Notationwise:
+a = ac * 10^aq
+b = bc * 10^bq
+etc.
+
+### OPCODE defs
+
+0xd0 DECADD a+b -> c    : (ac, aq, bc, bq, precision) -> (cc, cq)
+0xd1 DECNEG  -a -> b    : (ac, aq) -> (bc, bq)
+0xd2 DECMUL a*b -> c    : (ac, aq, bc, bq, precision) -> (cc, cq)
+0xd3 DECINV 1/a -> b    : (ac, aq, precision) -> (bc, bq)
+0xd4 DECEXP exp(a) -> b : (ac, aq, precision, steps) -> (bc, bq)
+0xd5 DECLN   ln(a) -> b : (ac, aq, precision, steps) -> (bc, bq)
+0xd6 DECSIN sin(a) -> b : (ac, aq, precision, steps) -> (bc, bq)
+
+precision is the # of digits kept during all calculations. steps for DECEXP and DECSIN are the # of Taylor expansion steps. steps for DECLN is the depth of the continued fractions expansion.
+
+### Why these functions?
+
+The proposed functions (+,-,*,/,exp,ln,sin) form a small set that combined enable all calculation of all elementary functions, which includes the sets of sums, products, roots and compositions of finitely many polynomial, rational, trigonometric, hyperbolic, and exponential functions, including their inverse functions.
+
+a^b = exp(b * ln(a)) gives us powers and polynomials.
+cos(a) = sin(tau/4-a), tan(a)=sin(a)/cos(a), etc., gives us all of trigonometry.
+
+together with arithmetic, we get all elementary functions.
+
+### DECNEG instead of DECSUB
+
+Negation is a more general operation vs subtraction. OPCODEs should be as fundamental as possible and as complex as desirable.
+For the same reason, we have DECINV instead of DECDIV.
+
+DECSUB(a,b) = DECADD(a,DECNEG(b))
+DECDIV(a,b) = DECMUL(a,DECINV(b))
+
+### DECEXP, DECSIN via Taylor series
+
+The Taylor series of exp and sin converge everywhere and fast. The error falls as fast as the factorial of steps.
+
+### DECLN via continued fractions
+
+Ln converges fast using continued fractions within the interval ]0,2]. The implementation scales the input into this interval and scales the result back correctly.
+
+### math/big
```

Comment:
renaming this section to Decimal arithmetic precision  or something would be better because thats what it talks  about

Reply:
removed math/big section (not really needed)